### PR TITLE
Chan

### PR DESCRIPTION
**Pull request by chanpham97**

### CODE REVIEW

File: script/db/index.js [hunk 1]
```diff
@@ -70,10 +70,12 @@ module.exports = firebase => {
     },
 
     /**
+<<<<<<< HEAD
```
Comment (alanhdu) [inline script/db/index.js]: Bad merge?

File: script/db/index.js [hunk 2]
```diff
@@ -138,7 +143,9 @@ module.exports = firebase => {
 
               const playListData = allPlaylists.val();
 
-              return userPlaylists.map(x => playListData[x].urls).every(currURLs => {
+              return userPlaylists.filter(x => x in playListData).map(x => {
+                return playListData[x].urls;
+              }).every(currURLs => {
```
Comment (alanhdu) [inline script/db/index.js]: This is a weird way to break up the lines. I'd prefer:

```javascript
return userPlaylists
  .filter(x => x in playListData)
  .map(x => playListData[x].urls)
  .every(currUrLS => {
    ...
  });
```

File: script/db/index.js [hunk 3]
```diff
@@ -161,10 +168,9 @@ module.exports = firebase => {
                 });
                 return playListKey;
               });
-              
             }
 
-            console.error(`ERROR: The user ${userName} already has this exact playList`);
+            console.error(`ERROR: The user ${ userName } already has this exact playList`);
```
Comment (alanhdu) [inline script/db/index.js]: Why the spaces?

File: script/routes/index.js [hunk 4]
```diff
@@ -48,17 +47,41 @@ module.exports = (app, db) => {
...
-      .catch(error => {
-        console.log(error.message);
-        res.redirect("/create");
+    if (req.session && req.session.user) {
+      console.log("in session");
+      const user = req.session.user;
+
+      db.readUserData(user.uid)
+        .then(userData => {
+          //TODO: don't include the password @Nick will change this
```
Comment (alanhdu) [inline script/routes/index.js]: What is this comment referring too?

File: script/routes/index.js [hunk 5]
```diff
@@ -48,17 +47,41 @@ module.exports = (app, db) => {
...
+      db.writePlaylistData("guest", "guest", title, description, fields)
+        .then(spid => {
+          console.log(spid);
+          res.redirect("/sp/" + spid);
+        })
+        .catch(error => {
+          console.log(error.message);
+          res.redirect("/create");
+        });
+    }
```
Comment (alanhdu) [inline script/routes/index.js]: I'd try to find a way to avoid duplicating this logic with lines 64-72.

File: script/db/index.js (hunk 1 above)
Comment (Howon) [inline script/db/index.js]: @chanpham97  fix it

File: script/db/index.js [hunk 6]
```diff
@@ -70,10 +70,12 @@ module.exports = firebase => {
     },
 
     /**
+<<<<<<< HEAD
      * @param  string uid of the user being sought after
      * @return dictionary the user data in the form of the tree structure
      */
-    readUserData: uid => {
+    readUserData: (uid) => {
```
Comment (Howon) [inline script/db/index.js]: unnecessary paren

**Comment by Howon** [comment]
Are you ever going to close this

**Pull request merged**